Extension seems to change current working directory to /tmp

MWE: 

Steps to reproduce

1. Download [test.zip](https://github.com/user-attachments/files/17416017/test.zip)
2. Works

```
jimisola@turing:~/cloud/dev/test/test (main)$mvn -Dnisse.source.jgit.dynamicVersion=true clean
[INFO] last commit: commit d0c593e5dccc16346b5d6bdc50b7fd9f58b7078f 1729181628 -----sp
[INFO] Nisse injecting 25 properties into User Properties
[INFO] Scanning for projects...
[INFO] Nisse inlining following properties:
[INFO]  * ${nisse.jgit.dynamicVersion}=0.0.1-1-SNAPSHOT
[INFO] Checking POMs for inlining:
[INFO]  * test:test needs inlining
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.1-1-SNAPSHOT
[INFO]   from .inlined7948929473423924619.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- clean:3.2.0:clean (default-clean) @ test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.222 s
[INFO] Finished at: 2024-10-17T18:21:49+02:00
[INFO] ------------------------------------------------------------------------

```

3.  sortpom without specifying pom.xml works

```
imisola@turing:~/cloud/dev/test/test (main)$mvn -Dnisse.source.jgit.dynamicVersion=true com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify
[INFO] last commit: commit d0c593e5dccc16346b5d6bdc50b7fd9f58b7078f 1729181628 -----sp
[INFO] Nisse injecting 25 properties into User Properties
[INFO] Scanning for projects...
[INFO] Nisse inlining following properties:
[INFO]  * ${nisse.jgit.dynamicVersion}=0.0.1-1-SNAPSHOT
[INFO] Checking POMs for inlining:
[INFO]  * test:test needs inlining
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.1-1-SNAPSHOT
[INFO]   from .inlined8619757747544758647.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sortpom:3.3.0:verify (default-cli) @ test ---
[INFO] Verifying file /tmp/.inlined8619757747544758647.xml
[INFO] The xml element <configuration> should be placed before <executions>
[INFO] The file /tmp/.inlined8619757747544758647.xml is not sorted
[INFO] Sorting file /tmp/.inlined8619757747544758647.xml
[INFO] Saved backup of /tmp/.inlined8619757747544758647.xml to /tmp/.inlined8619757747544758647.xml.bak
[INFO] Saved sorted pom file to /tmp/.inlined8619757747544758647.xml
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.302 s
[INFO] Finished at: 2024-10-17T18:27:20+02:00
[INFO] ------------------------------------------------------------------------
```

4. Speciyfing pom.xml in sortpom gives `/tmp/pom.xml`

```
jimisola@turing:~/cloud/dev/test/test (main)$mvn -Dnisse.source.jgit.dynamicVersion=true com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify -Dsort.pomFile="pom.xml"
[INFO] last commit: commit d0c593e5dccc16346b5d6bdc50b7fd9f58b7078f 1729181628 -----sp
[INFO] Nisse injecting 25 properties into User Properties
[INFO] Scanning for projects...
[INFO] Nisse inlining following properties:
[INFO]  * ${nisse.jgit.dynamicVersion}=0.0.1-1-SNAPSHOT
[INFO] Checking POMs for inlining:
[INFO]  * test:test needs inlining
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.1-1-SNAPSHOT
[INFO]   from .inlined12277631677121855567.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sortpom:3.3.0:verify (default-cli) @ test ---
[INFO] Verifying file /tmp/pom.xml
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.194 s
[INFO] Finished at: 2024-10-17T18:28:29+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.ekryd.sortpom:sortpom-maven-plugin:3.3.0:verify (default-cli) on project test: Could not read pom file: /tmp/pom.xml: /tmp/pom.xml (No such file or directory) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

5. Disable nisse extension be renaming mvn/extensions.xml and run 

```
jimisola@turing:~/cloud/dev/test/test (main)$mvn initialize 
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for test:test:jar:${nisse.jgit.dynamicVersion}
[WARNING] 'version' contains an expression but should be a constant. @ line 7, column 12
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test ${nisse.jgit.dynamicVersion}
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sortpom:3.4.0:sort (default) @ test ---
[INFO] Sorting file /home/jimisola/cloud/dev/test/test/pom.xml
[INFO] Pom file is already sorted, exiting
[INFO] 
[INFO] --- git-commit-id:5.0.0:revision (get-the-git-infos) @ test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.826 s
[INFO] Finished at: 2024-10-17T18:35:33+02:00
[INFO] ------------------------------------------------------------------------
```

6. Enable nisse extension again and see that git-commit-id:5.0.0:revision can't find a .git directory (which exists as shown)

```
jimisola@turing:~/cloud/dev/test/test (main)$mvn -Dnisse.source.jgit.dynamicVersion=true initialize 
[INFO] last commit: commit d0c593e5dccc16346b5d6bdc50b7fd9f58b7078f 1729181628 -----sp
[INFO] Nisse injecting 25 properties into User Properties
[INFO] Scanning for projects...
[INFO] Nisse inlining following properties:
[INFO]  * ${nisse.jgit.dynamicVersion}=0.0.1-1-SNAPSHOT
[INFO] Checking POMs for inlining:
[INFO]  * test:test needs inlining
[INFO] 
[INFO] -----------------------------< test:test >------------------------------
[INFO] Building test 0.0.1-1-SNAPSHOT
[INFO]   from .inlined11657034662024604772.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sortpom:3.4.0:sort (default) @ test ---
[INFO] Sorting file /tmp/.inlined11657034662024604772.xml
[INFO] Pom file is already sorted, exiting
[INFO] 
[INFO] --- git-commit-id:5.0.0:revision (get-the-git-infos) @ test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.515 s
[INFO] Finished at: 2024-10-17T18:37:18+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.github.git-commit-id:git-commit-id-maven-plugin:5.0.0:revision (get-the-git-infos) on project test: .git directory is not found! Please specify a valid [dotGitDirectory] in your pom.xml -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
jimisola@turing:~/cloud/dev/test/test (main)$ls .git
COMMIT_EDITMSG  config  HEAD  index  logs  objects  ORIG_HEAD  refs
jimisola@turing:~/cloud/dev/test/test (main)$git log
commit d0c593e5dccc16346b5d6bdc50b7fd9f58b7078f (HEAD -> main)
Author: Jimisola Laursen <jimisola@jimisola.com>
Date:   Mon Oct 7 22:56:17 2024 +0200

    Initial commit
```

**and if you run the last command with `-X` you'll see `[DEBUG]   (f) dotGitDirectory = /tmp/.git` which strengthens my theory that the extension changes current working directory to /tmp.**

--- a/extension3/src/main/java/eu/maveniverse/maven/nisse/extension3/internal/NissePropertyInliner.java
+++ b/extension3/src/main/java/eu/maveniverse/maven/nisse/extension3/internal/NissePropertyInliner.java
@@ -68,7 +68,7 @@ final class NissePropertyInliner {
                     Files.createDirectories(inlinedPomPath.getParent());
                     session.getRepositorySession().getData().set(INLINED_POM_PATH_KEY, inlinedPomPath);
                     inline(pomPath, inlinedPomPath, inlinedProperties);
-                    mavenProject.setFile(inlinedPomPath.toFile());
+                    mavenProject.setPomFile(inlinedPomPath.toFile());
                 }
             }
         } else {

